app/src/main/java/uc/edu/klopfsea/groupscheduler/model/Event.kt

```diff
@@ -16,12 +16,11 @@ data class Event (
     var year=0
     // Sets the time
     fun setTime(t: Int){
-        if (t==0)
-            timeSpecification = "After $time $period"
-        else if (t == 1)
-            timeSpecification = "Before $time $period"
-        else
-            timeSpecification = "All Day"
+        timeSpecification = when (t) {
```

Comment:
Changing this to when instead of an if else statement for this one seemed important due to it fitting the criteria to be a when instead. This doesn't change any functionality but instead it makes the code shorter and leads to cleaner more readable code here.  it is recommended to use when for these types of if else statements